checkers_bot_tournament/stat_printing.py

```diff
@@ -0,0 +1,210 @@
+
+from checkers_bot_tournament.bots.bot_tracker import BotTracker
+from checkers_bot_tournament.checkers_util import compute_performance_rating, make_unique_bot_string
+
+
+def write_tournament_overall_stats(bot_list: list[BotTracker], file: IO) -> None:
+    """
+    Writes game result statistics to a file in a structured and readable format.
+
+    The format includes a header row followed by counts and percentages for
+    White, Black, and Overall statistics for each bot.
+
+    Shamelessly crafted with ChatGPT :)
+
+    Args:
+        file: The file object to write the statistics to.
+        game_stats (Dict[str, GameResultStat]): A dictionary mapping bot names to their game statistics.
+    """
+    file.write("Game Statistics\n")
+    file.write("=" * 60 + "\n\n")
+
+    for bot in bot_list:
+        stats = bot.stats
+        file.write(f"Bot Name: {make_unique_bot_string(bot)} ({round(bot.rating)})\n")
+        file.write("-" * 60 + "\n")
+
+        label_width = 10  # For "White", "Black", "Overall"
+        col_width = 8  # For "Win", "Draw", "Loss" columns
+
+        # Print the header row once per bot
+        header = f"{'Win':<{col_width}}{'Draw':<{col_width}}{'Loss':<{col_width}}{'Overall Score':<{col_width}}"
+        file.write(f"{'':<{label_width}}{header}\n")
+
+        # Organize counts for White, Black, and Overall
+        counts = {
+            "White": (stats.white_wins, stats.white_draws, stats.white_losses),
+            "Black": (stats.black_wins, stats.black_draws, stats.black_losses),
+            "Overall": (stats.total_wins, stats.total_draws, stats.total_losses),
+        }
+
+        # Print absolute counts
+        for label, (w, d, l) in counts.items():
+            counts_str = f"{w:<{col_width}}{d:<{col_width}}{l:<{col_width}}"
+            file.write(f"{label:<{label_width}}{counts_str}\n")
+
+        # Calculate and print percentages
+        for label, (w, d, l) in counts.items():
+            total_games = w + d + l
+            if total_games > 0:
+                win_pct = (w / total_games) * 100
+                draw_pct = (d / total_games) * 100
+                loss_pct = (l / total_games) * 100
+                score_pct = ((w + 0.5 * d) / total_games) * 100
+            else:
+                win_pct = draw_pct = loss_pct = score_pct = 0.0
+
+            # Format each percentage including the % sign within the column
+            win_str = f"{win_pct:.1f}%"
+            draw_str = f"{draw_pct:.1f}%"
+            loss_str = f"{loss_pct:.1f}%"
+            score_str = f"{score_pct:.2f}%"
+
+            pct_str = (
+                f"{label:<{label_width}}"
+                f"{win_str:<{col_width}}"
+                f"{draw_str:<{col_width}}"
+                f"{loss_str:<{col_width}}"
+                f"= {score_str}"
+            )
+            file.write(pct_str + "\n")
+
+        file.write("=" * 60 + "\n\n")
+
+
+def write_tournament_h2h_stats(bot_list: list[BotTracker], file: IO) -> None:
```

Comment:
not that important but this do be a bit of a chonky function idk if it's worth splitting up